### ISSUE TITLE
ERC20 Custom Token state not saved to Ethereum Wallet / Mist

### ISSUE THREAD

**Issue by rbval** [ethereum/mist#2830]
``````
Version: `0.9.0`
OS & Version: osx
Node type: `geth`
Number of blocks synchronized: 4073091
``````

After the 0.9.0 update ERC20 Tokens seem to disappear from "Custom Tokens" every time Ethereum Wallet is restarted. It seems that Ethereum Wallet / Mist doesn't save the ERC20 token state properly.

I start the P2P network with the following command:
``````
geth --light
``````
After that I open the Ethereum Wallet / Mist normally with Spotlight. I could replicate the problem a few times.

**Comment by rondior** [comment]
I'm having the same issue.  Still havent found a solution.

**Comment by BigBearB** [comment]
I'm also losing the ERC20 contracts and tokens in wallet every time I shut down 0.9.0.
The tokens are still in my address on the blockchain, but this is crap.  Anyone have a solution, so I don't have to bring up the ERC20 contracts with JSON interface etc. each time I open the wallet ?

**Comment by tdpollan** [comment]
Same issue w/ 0.9.0 ethereum wallet not saving tokens.  Mist 0.9.0 works fine.

**Comment by evertonfraga** [comment]
We’re aware of the issue and working hard on a hotfix.

please hang tight :)

**Comment by jimmykl** [comment]
Ditto using geth --light

**Comment by GregoryChemin** [comment]
after realoding the ETH Wallet to most recent version, I could retrieve my ETH Balances, but ALL TOKEN, such as GOLEM, Monaco ... have ALL disappeared/Gone

Thought ???

**Comment by jimmykl** [comment]
@GregoryChemin Yes that's what we are all experiencing too. The tokens are still there, you can view them with a blockchain explorer, it's just the balance in Ethereum Wallet which isn't working. @evertonfraga says that a fix is being worked on.

**Comment by GregoryChemin** [comment]
@ Jimmykl, you are absultely right, i can retrieve them on Etherscan. Much appreciated

**Comment by evertonfraga** [comment]
Duplicate of #2823. Please follow updates on that issue.

**Comment by geronemo** [comment]
1. Bug - Restarting remove all tokens
2. Desireble (indespensible) feature - all tokens ahoukd be seen as default. Why I have to remember what I have and add it manually. in MEW I can see all tokens w/o any actions.

**Comment by tomtom87** [comment]
I am able to add my token in a private test net, but unable to sync to the testnets Ropsten or Rinkeby 100%, the last 100 blocks or so never download.

**Comment by sriharikapu** [comment]
I created erc20 tokens but they arent yetconfirmed since last 10 days althought i kept 3usd as mining fee on my native etherium wallet

**Comment by tomtom87** [comment]
How many confirmations on etherscan? Restart your node and sync again it should come through.

**Comment by sriharikapu** [comment]
thanks it worked finally

**Comment by jesobreira** [comment]
Same issue. My contracts (issued and/or watched) disappear once I restart the wallet.

**Comment by aphilsmith** [comment]
It's been 4 days for me. It went all the way down to zero blocks left, then restarted. The Mist wallet remained in a "creating contract" state, What can I do?

**Comment by BigBearB** [comment]
Dear Evertonfraga,

You are a genius.  Please, please, please make the wallet work for ERC20 without us having to use v. 8.10.
Please ?

**Comment by oslivan** [comment]
I upgrade to v0.9.2, the problem is still exist?

**Comment by geronemo** [comment]
looks like it resolved

**Comment by oslivan** [comment]
@geronemo thanks your reply, I need add a watch for token.

**Comment by realrover** [comment]
Upgraded to 0.9.2. The problem still exist under Win 10. Watched tokens are not shown. Contracts are OK. After some time (5-30 min) eventually tokens appear.

**Comment by Chris-Hindy** [comment]
I have the same problem on Mac 10.12.6 - tokens still not retained when I close the Wallet app and reopen later.... @realrover did you mean that the tokens will reappear after 5-30 minutes without you having to enter the token address again?

**Comment by DivikBansal** [comment]
I am not receiving ether issued from rinkeby faucet to my Ethereum wallet v9.2 although etherscan shows them being issued to main a/c

**Comment by toldenburger** [comment]
Same problem on MacOS 10.13, Ethereum Wallet 0.9.2, Geth 1.7.2
All tokens disappear after re-starting the app.

**Comment by 2infiniti** [comment]
Why is this never resolved? Its been absolute NIGHTMARE to re-add 20+ tokens on every re-launch! This has been occurring and never fixed since June! Currently under Ethereum Wallet 0.9.2 / Windows 10

**Comment by phobeus81** [comment]
I am also encountering this on linux system. This actually makes the wallet for token nearly completely useless.

**Comment by tomtom87** [comment]
Yeh it's only useful if you manage contracts otherwise use Myetherwallet 
or an exchange.

**Comment by chschm** [comment]
Issue still exists on my machine. Is it fixed in 0.9.3?

**Comment by fell0w** [comment]
I'm running 0.9.3 and I still have the issue. Added all the addresses for my custom token and all show 0.0000000 :(
On etherscan everything looks fine!

**Comment by Toldwone** [comment]
Hi there! Same issue, Ethereum Wallet is synced and up to date but I can't see any of my SUB...

**Comment by vickyzare567** [comment]
Same problem and also the tokens not showing on the geth node as well as etherscan ........
Need some help

**Comment by phobeus81** [comment]
If it is not on etherscan, it is something different and you might want to verify that you used the correct address. You might want to ask somewhere else for assistance understanding what happened.

However this issue is getting a major pain not had been fixed for month now, making the whole token system unusable.

**Comment by chschm** [comment]
It will show the tokens in Mist Wallet when they are added, but this is pretty game-breaking for the Ethereum Wallet

**Comment by jirupu** [comment]
I still have this problem. I was sent a considerable amount of REBL token for their airdrop but I can't find it. PLEASE I BEG YOU... Please fix this.. T_T

**Comment by weideng** [comment]
the same problem, Mac 10.13.2, wallet version 0.9.3, I can see my own tokens on etherscan, but still not shown in my wallet

**Comment by jirupu** [comment]
@weideng i find it impossible to show rebl on my mist wallet but found out that i can use the json file on mew and it works perfectly there. Or if u have the PK, you can access it there even if you created your wallet from mist. What token are you trying to track btw? Im using the rebl wallet now as i have rebl there and i find it perfect.

**Comment by cryptoalexwhite** [comment]
Having the same issue Mac 10.13.1 (17B1003) Ethereum Wallet 0.9.3. Etherscan shows the incoming transaction, but the wallet doesn't show it. Sync with Light client option is Enabled in the menu.

**Comment by efday** [comment]
The issue is still present in ethereum wallet  version 0.10.0 : i have to re-enter the tokens adresses each time i launch the wallet .
For info : geth version  1.8.2 , windows 8.1  , running wallet in light mode.

**Comment by kylegordon** [comment]
This does not appear to be an issue now in 0.11.0
